client/ayon_resolve/plugins/create/create_shot_clip.py

```diff
@@ -1,23 +1,136 @@
+
+        Args:
+            update_list(List[UpdateData]): Gets list of tuples. Each item
+                contain changed instance and it's changes.
+        """
+        for created_inst, _changes in update_list:
+            track_item = created_inst.transient_data["track_item"]
+            tag_data = lib.get_timeline_item_ayon_tag(track_item)            
+            instances_data = tag_data[_CONTENT_ID]
+
+            instances_data[self.identifier] = created_inst.data_to_store()
+            lib.imprint(track_item, tag_data)
+
+    def remove_instances(self, instances):
+        """Remove instance marker from track item.
+
+        Args:
+            instance(List[CreatedInstance]): Instance objects which should be
+                removed.
+        """
+        for instance in instances:
+            track_item = instance.transient_data["track_item"]
+            tag_data = lib.get_timeline_item_ayon_tag(track_item)
+            instances_data = tag_data.get(_CONTENT_ID, {})
+            _ = instances_data.pop(self.identifier, None)
+            self._remove_instance_from_context(instance)
+
+            # Remove markers if deleted all of the instances
+            if not instances_data: 
+                track_item.DeleteMarkersByColor(constants.ayon_marker_color)
+                if track_item.GetClipColor() != constants.selected_clip_color:
+                    track_item.ClearClipColor()
+
+            # Push edited data in marker
+            else:
+                lib.imprint(track_item, tag_data)
+
+
+class ResolveShotInstanceCreator(_ResolveInstanceCreator):
+    """Shot product type creator class"""
+    identifier = "io.ayon.creators.resolve.shot"
+    product_type = "shot"    
+    label = "Editorial Shot"
+
+
+class EditorialReviewInstanceCreator(_ResolveInstanceCreator):
+    """Review product type creator class"""
+    identifier = "io.ayon.creators.resolve.review"
+    product_type = "review"
+    label = "Editorial Review"
+
+
+class EditorialPlateInstanceCreator(_ResolveInstanceCreator):
+    """Plate product type creator class"""
+    identifier = "io.ayon.creators.resolve.plate"
+    product_type = "plate"
+    label = "Editorial Plate"
+
+
+class EditorialAudioInstanceCreator(_ResolveInstanceCreator):
+    """Audio product type creator class"""
+    identifier = "io.ayon.creators.resolve.audio"
+    product_type = "audio"
+    label = "Editorial Audio"
+
+
 class CreateShotClip(plugin.ResolveCreator):
     """Publishable clip"""
 
     identifier = "io.ayon.creators.resolve.clip"
     label = "Create Publishable Clip"
-    product_type = "clip"
+    product_type = "editorial"
     icon = "film"
     defaults = ["Main"]
 
-    create_allow_context_change = False
+#    create_allow_context_change = False  
+# TODO: explain consequence on folderPath
+# https://github.com/ynput/ayon-core/blob/6a07de6eb904c139f6d346fd6f2a7d5042274c71/client/ayon_core/tools/publisher/widgets/create_widget.py#L732
+
```

Comment:
If `create_allow_context_change = False` then the create instance_data does not contain the selected `folderPath`.
See: https://github.com/ynput/ayon-core/blob/6a07de6eb904c139f6d346fd6f2a7d5042274c71/client/ayon_core/tools/publisher/widgets/create_widget.py#L732

This looks like a bug to me, would you like me to create an issue for that ?
Is there a better work-around than simply making it editable ?

Reply:
Good question for @iLLiCiTiT - related to my comment here https://github.com/ynput/ayon-resolve/pull/15#discussion_r1743207592.

Reply:
If it is set to `False`, then it is job of create plugin to fill it.

Reply:
Thanks for the clarification.